apps/dgst.c

```diff
@@ -402,8 +402,9 @@ int dgst_main(int argc, char **argv)
 
     if (md == NULL) {
         EVP_MD_CTX *tctx;
+
         BIO_get_md_ctx(bmd, &tctx);
-        md = EVP_MD_CTX_md(tctx);
+        md = EVP_MD_CTX_get0_md(tctx);
```

Comment:
So here's a problem to keep track of: you must never try to free something you got from EVP_MD_CTX_get0_md()

Reply:
The same goes for an EVP_CIPHER returned by EVP_CIPHER_CTX_get0_cipher()

Reply:
yes, this was raised before.  And I said I have thoughts on how to handle it.

Reply:
Well, there's an obvious solution: use the get1 function instead